examples/platform/esp32/mode-support/static-supported-modes-manager.cpp

```diff
@@ -0,0 +1,173 @@
+#include "static-supported-modes-manager.h"
+#include "ESP32Config.h"
+#include <app/util/debug-printing.h>
+#include <app/util/ember-print.h>
+
+using namespace chip;
+using namespace chip::app::Clusters;
+using namespace chip::DeviceLayer::Internal;
+using namespace chip::app::Clusters::ModeSelect;
+using chip::Protocols::InteractionModel::Status;
+
+using ModeOptionStructType = Structs::ModeOptionStruct::Type;
+using SemanticTag          = Structs::SemanticTagStruct::Type;
+template <typename T>
+using List                                                                                = app::DataModel::List<T>;
+
+StaticSupportedModesManager::ModeLabel* StaticSupportedModesManager::modeLabelList = nullptr;
+StaticSupportedModesManager::ModeOptionStructType* StaticSupportedModesManager::modeOptionStruct = nullptr;
+StaticSupportedModesManager::SemanticTag* StaticSupportedModesManager::semanticTags = nullptr;
+
+const StaticSupportedModesManager StaticSupportedModesManager::instance = StaticSupportedModesManager();
+
+void *StaticSupportedModesManager::endpointArray[3][2];
```

Comment:
Where are these magic "3" and "2" numbers coming from?

What does this array actually store?  `void *` is not terribly descriptive.

Reply:
Used `ModeOptionStructType`